src/shogun/features/CombinedFeatures.cpp

```diff
@@ -162,6 +162,11 @@ void CombinedFeatures::init()
 {
 	SG_ADD(&num_vec, "num_vec", "Number of vectors.");
 	SG_ADD(&feature_array, "feature_array", "Feature array.");
+	add_callback_function(
+		"feature_array", [&]() { 
+			if (feature_array.size() > 0) 
+				num_vec = feature_array[0]->get_num_vectors(); 
+	});
 }
```

Comment:
when feature array is added, `num_vec` also needs to be changed, as get_num_vectors needs `num_vec`.
 https://github.com/shogun-toolbox/shogun/blob/a075792c536d67e295927577d12b7bf335565550/src/shogun/features/CombinedFeatures.h#L70-L75

Reply:
nice!
I think in fact num_vec should never be stored ... only every queried ...but this is ok as well. thx